### PR TITLE
fix the :Octo repo url command

### PR DESCRIPTION
**Pull request by Drew-Daniels**
### Describe what this PR does / why we need it
This PR fixes the `:Octo repo url` command. It silently fails when an Octo buffer is not opened, but this command should not require one to get the remote url.

### Does this pull request fix one issue?

Closes https://github.com/pwntester/octo.nvim/issues/628

### Describe how you did it
To fix, I've just used the utility functions `get_remote_host` and `get_remote_name` and concatenation to build out the remote url name, which can be used to get the remote url when an Octo buffer is/is not opened.

#### When not in a Git repo
- `cd` to a local directory that is not a `git` repository:
- open `neovim`
- run `:Octo repo url`
- verify you see the "No remote repository found" message

#### When in a Git repo that is _not_ tracking a GitHub repository:
- `cd` to a local `git` repository that is tracking a remote GitHub repository:
- open `neovim`
- run `:Octo repo url`
- verify you see the "No remote repository found" message

#### When in a Git repo that _is_ tracking a remote GitHub repository
- `cd` to a local `git` repository that _is_ tracking a remote GitHub repository:
- open `neovim`
- run `:Octo repo url`
- verify you see the "Copied URL <> to the system clipboard (+ register)" message
- verify that when you paste, you see the remote url as contents

### Checklist

- [ ] Passing tests and linting standards
- [ ] Documentation updates in README.md and doc/octo.txt

### CODE REVIEW

File: lua/octo/commands.lua [hunk 1]
```diff
@@ -1669,10 +1669,20 @@ end
...
+  if buffer then
+    url = buffer.node.url
+  else
+    local host = utils.get_remote_host()
+    local remote_name = utils.get_remote_name()
+    if not host or not remote_name then
+      utils.error "No remote repository found"
+      return
+    end
+    url = "https://" .. host .. "/" .. remote_name
```
Comment (wd60622) [inline lua/octo/commands.lua:1683]: Might be a good function for utils `get_remote_url`

**Review by wd60622 — COMMENTED** [review]
Thanks for picking this up @Drew-Daniels 

Small suggestion

File: lua/octo/commands.lua (hunk 1 above)
Comment (Drew-Daniels) [reply lua/octo/commands.lua]: Ah I agree. Just updated!

**Review by wd60622 — APPROVED** [review]
Works for me! Thanks for the edit

**Pull request merged**